### PR TITLE
refactor(router): subscribe router events until destroyed

### PR DESCRIPTION
**Pull request by JLHwung**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

If `setUpPreloading` is not called, the `ngOnDestroy` lifecycle will throw `can not read property `unsubscribe` of undefined.

## What is the new behavior?
It should not throw this error.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

The `RouterPreloader` now does not relies on external `setUpPreloading` calls.

### CODE REVIEW

File: packages/router/src/router_preloader.ts [hunk 1]
```diff
@@ -85,21 +84,21 @@ export class RouterPreloader implements OnDestroy {
   }
 
   setUpPreloading(): void {
-    this.subscription =
-        this.router.events
-            .pipe(filter((e: Event) => e instanceof NavigationEnd), concatMap(() => this.preload()))
-            .subscribe(() => {});
+    this.router.events
+        .pipe(filter((e: Event) => e instanceof NavigationEnd), concatMap(() => this.preload()))
+        .pipe(takeUntil(this.destroyed))
```
Comment (alfaproject) [inline packages/router/src/router_preloader.ts]: Why 2 `pipe`?

**Comment by alfaproject** [comment]
Should be easy to make a unit test for this and I think it's required to be able to be merged at some point

File: packages/router/src/router_preloader.ts (hunk 1 above)
Comment (JLHwung) [reply packages/router/src/router_preloader.ts]: Ah, I could have used only one pipe. Fixed.

**Comment by alfaproject** [comment]
#1 is what you fixed right? That's probably good enough

**Review by benlesh — CHANGES_REQUESTED** [review]
Using a Subject here adds a bit of unnecessary overhead. It's a fine approach when you have more than one way to terminate a subscription, or you're managing a lot of subscriptions.

In this case, I think we'd be fine just checking to see if `this.subscription` exists, and calling `this.subscription.unsubscribe()` accordingly. That's the least overhead and the fewest allocations.

(Unrelated: The call to `complete` on the destroyed Subject is superfluous in this case)

**Comment by JLHwung** [comment]
@benlesh Revised with simpler empty check for `this.subscription`.

**Comment by atscott** [comment]
Looks like this was addressed in https://github.com/angular/angular/commit/763023472b15f82a7551c9c4f8f43cda1d259dd5. Sorry about this change getting lost!

**Pull request closed without merging**